src/pybind/mgr/cephadm/serve.py

```diff
@@ -888,6 +888,13 @@ def update_progress() -> None:
                         hosts_altered.add(d.hostname)
                         break
 
+                # do not attempt to deploy node-proxy agent when oob details are not provided.
+                if slot.daemon_type == 'node-proxy' and not slot.hostname in self.mgr.node_proxy_cache.oob.keys():
```

Comment:
flake8 failure
```
cephadm/serve.py:892:57: E713 test for membership should be 'not in'
1     E713 test for membership should be 'not in'
```
it wants `and slot.hostname not in self.mgr.node_proxy_cache.oob.keys()` instead

Reply:
doh! true 🙂